src/selectors/getAllProjectKeys.js

```diff
@@ -1,10 +1,26 @@
 import {createSelector} from 'reselect';
+import isNull from 'lodash-es/isNull';
 
 export default createSelector(
   state => state.get('projects'),
   projects => projects.
-    filter(project => project.updatedAt).
-    sortBy(project => -project.updatedAt).
+    sort((...args) => {
+      const [firstProject, secondProject] = args;
+      const firstProjectUpdatedAt = firstProject.updatedAt;
+      const secondProjectUpdatedAt = secondProject.updatedAt;
```

Comment:
Ack I’m sorry @jwang1919 I totally missed this—so there is some destructuring going on here but it was not what I had in mind. Generally we want to use destructuring to make the code more concise, avoid repetition, and make explicit what parts of an object’s shape we care about. So in this case, since we only care about the `updatedAt` field, we can destructure right in the function params and avoid three intermediate variables:

```js
sort(({updatedAt: firstProjectUpdatedAt}, {updatedAt: secondProjectUpdatedAt}) => {
  const isFirstProjectNull = isNull(firstProjectUpdatedAt);
  // etc.
});
```

Reply:
Hmm.. didn't know I can do it this way. Sure can.